lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -446,33 +455,49 @@ LogicalResult ControlMergeOp::inferReturnTypes(
   // ControlMerge must have at least one data operand
   if (operands.empty())
     return failure();
-  // Result type is type of any data operand and index type
+  // Result type is type of any data operand and, by default, an index type
   inferredReturnTypes.push_back(operands[0].getType());
   inferredReturnTypes.push_back(IndexType::get(context));
   return success();
 }
 
 ParseResult ControlMergeOp::parse(OpAsmParser &parser, OperationState &result) {
   SmallVector<OpAsmParser::UnresolvedOperand, 4> allOperands;
-  Type type;
-  ArrayRef<Type> operandTypes(type);
+  Type resultType, indexType;
+  ArrayRef<Type> operandTypes(resultType);
   SmallVector<Type, 1> resultTypes, dataOperandsTypes;
```

Comment:
As you are already changing this: Could you drop the fixed size of 1, it's not the correct size, so we better drop it.